mate-screenshot/src/mate-screenshot.c

```diff
@@ -851,7 +855,15 @@ finish_prepare_screenshot (char *initial_uri, GdkWindow *window, GdkRectangle *r
 
   play_sound_effect (window);
 
-  dialog = screenshot_dialog_new (screenshot, initial_uri, take_window_shot);
+  if (noninteractive_clipboard_arg) {
+    save_screenshot_in_clipboard (gdk_window_get_display (window), screenshot);
+    g_free (initial_uri);
+    /* Done here: */
+    gtk_main_quit ();
+    return;
+  } else {
+    dialog = screenshot_dialog_new (screenshot, initial_uri, take_window_shot);
+  }
```

Comment:
This block feels strange, if you're wrapping the noninteractive clipboard above in a conditional and then quitting, why not either wrap the `dialog` part entirely, or fall through without an else? As it is, you have the `dialog = ...` part divided between an else and a fall through, but it will only ever work for the else anyway.